mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -85,6 +86,10 @@ abstract class Predictor[
     M <: PredictionModel[FeaturesType, M]]
   extends Estimator[M] with PredictorParams {
 
+  protected[spark] var storageLevel = StorageLevel.NONE
+
+  protected def handlePersistence = storageLevel == StorageLevel.NONE
```

Comment:
This design may cause some concurrency issue I think.
For example, if we want to use the predictor to train different dataset parallelly, the `storageLevel` status will be in a mess.

Reply:
Thanks a lot for reviewing this!
I am OK to revert `Predictor` and define `handlePersistence` in each algs. However, I don't quite understand `train different dataset parallelly`, do you mean using multi-threads to call `fit` in single instance?

Reply:
Yes... so it looks like a change breaking thread safety.
and `handlePersistence` is information related to `Dataset`, but it is add as a member of `Predictor`, I think the design is a little weird and maybe will cause other potential bugs.

Reply:
+1 @WeichenXu123 We can't store persistence level as the member variable of _Predictor_, since we may use one estimator to fit on the same dataset multiple times, and it may lead duplicated persistence if we don't handle it carefully. Thanks.